Mathlib/Order/InitialSeg.lean

```diff
@@ -198,6 +218,48 @@ protected theorem acc (f : r ≼i s) (a : α) : Acc r a ↔ Acc s (f a) :=
     obtain ⟨a', rfl⟩ := f.mem_range_of_rel hb
     exact ha _ (f.map_rel_iff.mp hb), f.toRelEmbedding.acc a⟩
 
+section PartialOrder
+
+variable [PartialOrder β] {a a' : α} {b : β}
+
+theorem mem_range_of_le [Preorder α] (f : α ≤i β) (h : b ≤ f a) : b ∈ Set.range f := by
+  obtain rfl | hb := h.eq_or_lt
+  exacts [⟨a, rfl⟩, f.mem_range_of_rel hb]
+
+@[simp]
+theorem le_iff_le [PartialOrder α] (f : α ≤i β) : f a ≤ f a' ↔ a ≤ a' :=
+  f.toOrderEmbedding.le_iff_le
+
+@[simp]
+theorem lt_iff_lt [PartialOrder α] (f : α ≤i β) : f a < f a' ↔ a < a' :=
+  f.toOrderEmbedding.lt_iff_lt
```

Comment:
Can you check gcongr proves the <- direction? If it doesn't, you might need to add an `OrderHomClass` instance

Reply:
I added the `OrderHomClass` instance, but this still makes no progress:
```
example [PartialOrder α] (f : α ≤i β) (h : a ≤ a') : f a ≤ f a' := by gcongr
```

Reply:
I added the `OrderHomClass` instance, but this still makes no progress:
```lean 4
example [PartialOrder α] (f : α ≤i β) (h : a ≤ a') : f a ≤ f a' := by gcongr
```

Reply:
Did you happen to add it in #17224? Because that PR is open :P

Reply:
Ah damn!